### PR TITLE
WIP: Generate deployment files fom Helm charts

### LINKED ISSUE #143: static manifests from helm chart
Currently we have two different implementations of the driver plumbing. The helm chart and the static manifests. They can and do get out of sync. We could generate the static manifests as part of the release process by running 'helm template' on the chart. Then there would only be one set of templates to update for both use cases.

### PR DESCRIPTION
**Pull request by wilmardo**
# TODO:

- [x] Rebase after #570 is merged
- [ ] Rewrite E2E tests
- [ ] Rewrite documentation
- [ ] Rewrite examples
- [ ] Adds readme for deployment files ([example](https://github.com/keel-hq/keel/tree/master/deployment))
- [ ] Look into rook dependency on deployment files

# Describe what this PR does #

Followup of #570 and fixes #143

Generate deployment files to keep the maintenance of changes in one place (the Helm chart) to avoid errors and changed not being implement in one of the versions as we saw with #555 
A single source of truth :)

Also generated deployments deploy to the ceph-csi namespace which is created on deployment. Bonus is that the generated templates by Helm are now linted by yamllint :)

### TLDR
* Deployment files in `/deploy` now generated
* Generated deployments deploy to ceph-csi namespace instead of default
* Generated deployment files are linted by yamllint which also brings up errors in the Helm template indenting and styling.

## Is there anything that requires special attention ##

* The deploy files are now splitted by the template names in the helm chart which is different from the previous split by function(?).
* The topoly.enabled in the Helm chart now defaults to true to generate the same deployments files as commited in #816. In the PR the static manifests enable the topology feature gate without domains. But the Helm chart defaults to a disabled topology and the feature gate isn't enabled.
Now it is consistent, by default the feature gate gets enabled without domains.
* Podsecurity is default enabled in the Helm charts now, simplifies the generation and E2E tests, no need to override the values each time. I see no problem with this since in 1.14 psp was already beta and default enabled:
https://v1-14.docs.kubernetes.io/docs/concepts/policy/pod-security-policy/
* Default all the tests to use the Helm chart since this is the new source of truth. For the sake of completeness the static manifests are tested against the latest k8s version only.

# How to test this PR / New workflow

* Make a change to one of the templates in the `charts/` directory
* Run `./scripts/gen-deploy.sh` to generate the deployment files.
* Run `./scripts/lint-text.sh` to lint all the newly generated files and the changes in the Helm chart

### CODE REVIEW

**Comment by nixpanic** [comment]
I would love to see this in release-3.0, @Madhu-1 what do you think?

**Comment by Madhu-1** [comment]
@nixpanic this is one of the long pending improvement. @wilmardo thanks again for taking time to work on this

**Comment by wilmardo** [comment]
Will rewrite the OP of this PR once I am up to speed with the changes again. Not everything is relevant anymore (1.13 is removed for example).

Will pick this up soon, today I rebased again master and setup the generation with `--show-only`. 
Next on the agenda are the E2E tests :)

File: .travis.yml [hunk 1]
```diff
@@ -27,7 +27,11 @@ env:
     - TEST_COVERAGE=stdout
     - GO_METALINTER_THREADS=1
     - GO_COVER_DIR=_output
+<<<<<<< HEAD
     - HELM_VERSION=v3.1.2
+=======
+    - HELM_VERSION=v3.2.0-rc.1
+>>>>>>> fix: set to Helm3 in CI
```
Comment (nixpanic) [inline .travis.yml:34]: this will likely break things, weird that Travis CI passed!?

Comment (nixpanic) [reply .travis.yml:34]: Ah, it did not even run! Probably because of this YAML error.

**Comment by Madhu-1** [comment]
@wilmardo hey any update on this one?

**Comment by Madhu-1** [comment]
moved this one out of release-v3.0.0 milestone

**Comment by wilmardo** [comment]
@Madhu-1 Sorry for this to linger so long, my time for opensource has been sparse lately.

@nixpanic @Madhu-1 Is there a convenient way to test the E2E test like the CI does locally? I tried running `scripts/travis-helmtest.sh v1.18.5 --test-cephfs=false --test-rbd=true` and even with some adaption to use the docker driver for minikube it wouldn't work. Before I dig into this further, how do you guys dev locally?

**Comment by Madhu-1** [comment]
> @Madhu-1 Sorry for this to linger so long, my time for opensource has been sparse lately.
> 
> @nixpanic @Madhu-1 Is there a convenient way to test the E2E test like the CI does locally? I tried running `scripts/travis-helmtest.sh v1.18.5 --test-cephfs=false --test-rbd=true` and even with some adaption to use the docker driver for minikube it wouldn't work. Before I dig into this further, how do you guys dev locally?
> 
I use VM_DRIVER=kvm2 as extra i dont use docker driver, nothing much 

> Also made it a draft since that is now possible with existing PR's :)
